### DISCUSSION TITLE
Apps depending on other apps (microservices)

### DISCUSSION THREAD

**Discussion by mars3142** [dokku/dokku#7265] [Ideas]
I would like to create a microservice platform on my VPS with dokku. The problem I'm facing right now is, the fact, that a reboot will break my microservices, because they need to be started in a specific order (discovery -> config -> others....).

I search the documentation, but couldn't find any "depends on" setting for apps. This is possible with docker compose, but how can I achieve it with dokku?

This would a cool extension for the project or is the correct way to use the "restart policy" and fail fast, if the other service is not running?

**Comment by josegonzalez** [comment]
There isn't any way to "depend" on a service per-se. Compose has this because of how compose starts containers, while Dokku starts them separetly via `docker run`.

Really what you should be doing is architecting your apps such that if downstream services arent working, they respond with the correct status codes and just retry connections until they come up (probably with exponential backoff somehow?). Otherwise you might get into a state where some downstream service has a minor blip and suddenly causes your entire setup to restart and issue a thundering herd of requests to other systems (such as databases or other "dependent" services).

I don't think trying to auto-restart downstream dependencies would be easy, but also don't think its a good idea (at least based on my experience managing infra).
